enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,227 @@
+Cluster manager will continue to provision `EgressIP` to nodes. Currently, it only assigns them to nodes which have a
+well-known label egress assignable `k8s.ovn.org/egress-assignable`.
+
+This enhancement is proposing enhancing cluster manager if the egress IP is within a subnet of a known non-primary network, then it
+will also consider which node hosts this secondary network. If no secondary network is found, we continue the default
+behaviour of assigning the egress IP to the primary network like what we do today.
+
+Cluster manager will therefore need to understand all available networks within the cluster and this enhancement proposes
+to add all secondary interfaces addresses and subnet mask to existing annotation `k8s.ovn.org/host-addresses` which is
+added to all node custom resource objects.
+
+Egress IP must not equal an address already assigned to an interface, and we can use the aforementioned `k8s.ovn.org/host-addresses`
+annotation to detect this scenario.
+
+With a prerequisite that subnets must not overlap, we can determine which interface can host an egress IP.
+We will use the longest prefix match to determine which interface the egress IP is assigned on.
+
+To allow packets to be steered into secondary networks, OVN reroute logical policies on `ovn_cluster_router` will redirect
+the packet out to the management port of the node which the egress IP is hosted.
+
+We will add rules in linux routing policy database control for each source Pod IP and lookup the correct routing table
+which will contain one route - route to the correct interface.
+
+We will also add a masquerade rule to iptables when a packet matching the source pod IP and also is leaving
+a particular interface.
+
+
+See section `Design Details` for full implementation details.
+
+### User Stories
+As a user, I want the ability for cluster workload traffic to be routed to a network that is not managed by OVN and to have a consist source IP
+that is configurable.
+
+### Test Plan
+E2E tests verifying the correct source IP for deployment modes global IC and multi-zone IC:
+- Configure new EgressIP both on primary and secondary networks
+- Migrate existing EgressIP on primary and secondary networks
+- Delete existing EgressIP on primary and secondary networks
+- Add node
+- Delete node
+- Edit node
+- Add pod
+- Delete pod
+- Remove pod
+- Delete a valid rule manually 
+- Delete a valid IPtables rule manually
+- Delete a valid route manually
+- Delete a valid egress IP assigned to an interface manually
+- Valid egress service and egress IP functionality if both are active on the same k8 objs.
+Today, if egress service and egress IP both 'match' on the same pod, egress service is preferred.
+Continue this for this enhancement where egress service and egress IP overlap, prefer egress service.
+
+TBD: Should we have to consider, and therefore test NICs hot plugging / removal?
```

Comment:
maybe not hot plugging, but maybe changing / creating / deleting secondary networks that would affect the eip validation. Including changes of mask size that affect longest prefix match